File: Java/src/Valid Parentheses .java
public class Solution {
    public boolean isValid(String s) {
        char[] cString = s.toCharArray();
        Map<Character, Character> bracksets = new HashMap<>();
        bracksets.put('(', ')');
        bracksets.put('{', '}');
        bracksets.put('[', ']');
        
        Stack<Character> stack = new Stack<>();
        for (char c : cString){
            if (bracksets.containsKey(c)){
                stack.push(bracksets.get(c));
            }
            else if(stack.isEmpty() || c!= stack.pop() ){
               return false;
            }
        }
        return stack.isEmpty();
    }
}

File: Python/Valid Parentheses .py
class Solution:
    # @return a boolean
    def isValid(self, s):
        bracksets = {')':'(',']':'[','}':'{'}
        
        
        if not s:
            return False
        charlist = []
        for c in s:
            if c == '(' or c == '{' or c== '[':
                charlist.append(c)
            if c == ')' or c == '}' or c== ']':
                if not charlist:
                    return False
                if charlist[-1] == bracksets[c]:
                     charlist.pop()
                else:
                    return False
        return False if charlist else True